club.haroohie.SerialLoops.yaml

```diff
@@ -0,0 +1,89 @@
+id: club.haroohie.SerialLoops
+runtime: org.kde.Platform
+runtime-version: '6.9'
+base: net.kuribo64.melonDS
+base-version: stable
+sdk: org.kde.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.dotnet8
+  - org.freedesktop.Sdk.Extension.llvm20
+build-options:
+  prepend-path: "/usr/lib/sdk/dotnet8/bin"
+  append-ld-library-path: "/usr/lib/sdk/dotnet8/lib"
+  prepend-pkg-config-path: "/usr/lib/sdk/dotnet8/lib/pkgconfig"
+
+command: SerialLoops
+
+finish-args:
+  - --device=dri
+  - --socket=x11
+  - --share=ipc
+  - --socket=pulseaudio
+  - --filesystem=~/SerialLoops:create
+  - --env=DOTNET_ROOT=/app/lib/dotnet
+  - --env=SL_LLVM_PATH=/app/lib/llvm20
+  - --env=SL_NINJA_PATH=/app/bin/ninja
+  - --env=SL_BUNDLED_EMULATOR=/app/bin/melonDS
+  - --env=SL_STORE_SYS_CONFIG=false
+  - --env=SL_USE_UPDATER=false
+
+modules:
+  - name: dotnet
+    buildsystem: simple
+    build-commands:
+    - /usr/lib/sdk/dotnet8/bin/install.sh
+
+  - name: llvm
+    buildsystem: simple
+    build-commands:
+    - mkdir -p /app/lib/llvm20/bin/
+    - mkdir -p /app/lib/llvm20/lib/
+    - cp /usr/lib/sdk/llvm20/bin/clang /app/lib/llvm20/bin
+    - cp /usr/lib/sdk/llvm20/bin/ld.lld /app/lib/llvm20/bin
+    - cp /usr/lib/sdk/llvm20/bin/llvm-objcopy /app/lib/llvm20/bin
+    - cp /usr/lib/sdk/llvm20/bin/llvm-objdump /app/lib/llvm20/bin
+    - cp -r /usr/lib/sdk/llvm20/lib/* /app/lib/llvm20/lib/
+
+  - name: Ninja
+    buildsystem: simple
+    sources:
+      - type: git
+        url: https://github.com/ninja-build/ninja.git
+        tag: v1.13.1
+    build-commands:
+      - python3 ./configure.py --bootstrap
+      - mv ninja ${FLATPAK_DEST}/bin/ninja
+
+  - name: SDL
```

Comment:
2.23.4 is in the runtime. isn't that enough?

Reply:
Yep, that should be sufficient. Only reason we weren't already using it was because I wasn't aware it was available! Will switch now.

Reply:
Then it should be fixed. `pkg-config` should find sdl2, and if they don't use that it's a bug.

Reply:
Okay, I'll fork it and publish my own version.

Reply:
Okay, I have looked into this more. The library does not contain a bug -- it is simply using the default .NET native library loader. See [here](https://learn.microsoft.com/en-us/dotnet/core/dependency-loading/default-probing) and [here](https://learn.microsoft.com/en-us/dotnet/standard/native-interop/native-library-loading) for more information, but suffice it to say that it is looking for `libSDL2.so` which does appear to be present in the final flatpak sandbox, despite being present in `org.kde.Sdk`. Specifically, these two files are present in the x86_64 image:
```
/usr/lib/x86_64-linux-gnu/libSDL2-2.0.so.0
/usr/lib/x86_64-linux-gnu/libSDL2-2.0.so.0.3200.4
```
But `/usr/lib/x86_64-linux-gnu/libSDL2.so` is not present, so as a result, it's not resolving the library.

Reply:
I wrote a custom DLL importer that replaces the search for `libSDL` with a specific search for `libSDL2-2.0.so.0` and the application now functions as intended. To me this seems like a bug in the app builder, but if this is intended behavior I can commit with the custom DLL importer.

Reply:
if it's looking for a `.so` then it's doing it wrong. Whoever implemented it doesn't understand .so versions. They are versioned and it should be looking for the versioned file.

I most Linux distro the `.so` is only part of the development pacakges. They are symlinks to the actually version you'll link against and the linker understand that.

Reply:
Okay. I have put in a fix on the application side.